### PR TITLE
Add softmax versions of *categorical_crossentropy objectives

### PR DESCRIPTION
**Pull request by naure**
#2532

And clarifications in doc strings
Allow the use of the integrated softmax+crossentropy from Tensorflow without extraneous conversions

### CODE REVIEW

File: keras/activations.py [hunk 1]
```diff
@@ -3,16 +3,7 @@
...
-    if ndim == 2:
-        return K.softmax(x)
-    elif ndim == 3:
-        e = K.exp(x - K.max(x, axis=-1, keepdims=True))
-        s = K.sum(e, axis=-1, keepdims=True)
-        return e / s
-    else:
-        raise Exception('Cannot apply softmax to a tensor that is not 2D or 3D. ' +
-                        'Here, ndim=' + str(ndim))
+    return K.softmax_3d(x)
```
Comment (fchollet) [inline keras/activations.py]: Moving this around forces you to have one copy in each backend. Not good!

File: keras/backend/theano_backend.py [hunk 2]
```diff
@@ -721,13 +721,28 @@ def softmax(x):
...
+
+
 def softplus(x):
     return T.nnet.softplus(x)
 
 
 def categorical_crossentropy(output, target, from_logits=False):
     if from_logits:
-        output = T.nnet.softmax(output)
+        output = softmax_3d(output)
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: `T.nnet.softmax` creates less ops and will be more efficient (also simpler).

**Comment by fchollet** [comment]
> Add softmax versions of *categorical_crossentropy objectives

That in itself seems like a reasonable change, but I don't get the other changes.

**Comment by naure** [comment]
The backend softmax functions expect 2d tensors, so the objective function doesn't work for sequences in theano. This is why the existing activation function uses this 3d fallback, which I'm trying to reuse in the backend modules.
